collections/src/main/scala/strawman/collection/immutable/ChampHashSet.scala

```diff
@@ -0,0 +1,607 @@
+package strawman
+package collection
+package immutable
+
+import mutable.{Builder, ImmutableBuilder}
+import Hashing.computeHash
+
+import scala.{Any, AnyRef, Array, Boolean, Double, IndexOutOfBoundsException, Int, NoSuchElementException, SerialVersionUID, Serializable, Unit, `inline`, sys, Some, None}
+import scala.Predef.{assert, require, ???}
+
+import java.lang.Integer.bitCount
+import java.lang.Math.ceil
+import java.lang.System.arraycopy
+
+/** This class implements immutable sets using a Compressed Hash-Array Mapped Prefix-tree.
+  * See paper https://michael.steindorfer.name/publications/oopsla15.pdf for more details.
+  *
+  *  @tparam A      the type of the elements contained in this hash set.
+  *
+  *  @author  Michael J. Steindorfer
+  *  @version 2.13
+  *  @since   2.13
+  *  @define Coll `immutable.ChampHashSet`
+  *  @define coll immutable champ hash set
+  */
+@SerialVersionUID(2L)
+private[immutable] sealed case class ChampHashSet[A](val rootNode: SetNode[A], val cachedJavaHashCode: Int, val cachedSize: Int)
+  extends Set[A]
+    with SetOps[A, ChampHashSet, ChampHashSet[A]]
+    with StrictOptimizedIterableOps[A, ChampHashSet, ChampHashSet[A]]
+    with Serializable {
+
+  def iterableFactory = ChampHashSet
```

Comment:
Would you mind adding an explicit type annotation?

~~~ scala
def iterableFactory: IterableFactory[ChampHashSet] = ChampHashSet
~~~

Reply:
Done.